src/olympia/accounts/tests/test_views.py

```diff
@@ -1123,6 +1124,49 @@ def test_picture_upload_valid(self):
             'picture_upload': [u'Images must be either PNG or JPG.']}
 
 
+class TestAccountViewSetDelete(TestCase):
+    client_class = APITestClient
+
+    def setUp(self):
+        self.user = user_factory()
+        self.url = reverse('account-detail',
+                           kwargs={'pk': self.user.pk})
+        super(TestAccountViewSetDelete, self).setUp()
+
+    def test_delete(self):
+        self.client.login_api(self.user)
+        response = self.client.delete(self.url)
+        assert response.status_code == 204
+        assert self.user.reload().deleted
+
+    def test_no_auth(self):
+        response = self.client.delete(self.url)
+        assert response.status_code == 401
+
+    def test_different_account(self):
+        self.client.login_api(self.user)
+        url = reverse('account-detail', kwargs={'pk': user_factory().pk})
+        response = self.client.delete(url)
+        assert response.status_code == 403
+
+    def test_admin_patch(self):
+        self.grant_permission(self.user, 'Users:Edit')
+        self.client.login_api(self.user)
+        random_user = user_factory()
+        url = reverse('account-detail', kwargs={'pk': random_user.pk})
+        response = self.client.delete(url)
+        assert response.status_code == 204
+        assert random_user.reload().deleted
+
+    def test_developers_cant_delete(self):
+        self.client.login_api(self.user)
+        addon_factory(users=[self.user])
+        response = self.client.delete(self.url)
+        assert response.status_code == 400
+        assert 'You must delete, or transfer' in response.content
+        assert not self.user.reload().deleted
```

Comment:
Can we include a test that tests for themes too? I noticed that even a little typo, checking for `is_addon_developer` instead of `is_developer` changes this behavior drastically.

Also a test for deleted addons (ignored by `is_developer`) would be nice so that we can better reason about the behavior of the API here too, e.g if someone changes that.